fix-point/src/CopyFiles.hs

```diff
@@ -0,0 +1,48 @@
+module CopyFiles where
+
+import Control.Applicative
+import Control.Exception (throw)
+import Control.Monad (forM_, unless, when)
+import Data.Text (pack)
+import System.Directory (copyFile, createDirectory, doesDirectoryExist,
+                         doesFileExist, listDirectory)
+import System.FilePath ((</>), takeExtension)
+
+import qualified Rename as R
+
+copyAll :: FilePath -> FilePath -> String -> IO ()
+copyAll source target newName = do
+    unlessM (doesDirectoryExist source) $
+        throw (userError "source does not exist")
+    whenM (doesFileOrDirectoryExist target) $
+        throw (userError "destination already exists")
+
+  -- if bottom two lines swapped, infinite creation of target occurs
+    content <- listDirectory source
```

Comment:
I think you can write this code slightly shorter if you write something like this:

```haskell
content <- filter (/= "fix-point") <$> listDirectory source
```

-1 variable name and -1 line of code 👍